# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContextVertexArray.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLContextVertexArray.cpp
--- a/dom/canvas/WebGLContextVertexArray.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLContextVertexArray.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -13,8 +13,9 @@
 namespace mozilla {
 
 void WebGLContext::BindVertexArray(WebGLVertexArray* array) {
-  const FuncScope funcScope(*this, "bindVertexArray");
+  FuncScope funcScope(*this, "bindVertexArray");
   if (IsContextLost()) return;
+  funcScope.mBindFailureGuard = true;
 
   if (array && !ValidateObject("array", *array)) return;
 
@@ -28,28 +29,15 @@
   if (mBoundVertexArray) {
     mBoundVertexArray->mHasBeenBound = true;
   }
+
+  funcScope.mBindFailureGuard = false;
 }
 
-already_AddRefed<WebGLVertexArray> WebGLContext::CreateVertexArray() {
+RefPtr<WebGLVertexArray> WebGLContext::CreateVertexArray() {
   const FuncScope funcScope(*this, "createVertexArray");
   if (IsContextLost()) return nullptr;
 
-  RefPtr<WebGLVertexArray> globj = CreateVertexArrayImpl();
-  return globj.forget();
-}
-
-WebGLVertexArray* WebGLContext::CreateVertexArrayImpl() {
   return WebGLVertexArray::Create(this);
 }
 
-void WebGLContext::DeleteVertexArray(WebGLVertexArray* array) {
-  const FuncScope funcScope(*this, "deleteVertexArray");
-  if (!ValidateDeleteObject(array)) return;
-
-  if (mBoundVertexArray == array)
-    BindVertexArray(static_cast<WebGLVertexArray*>(nullptr));
-
-  array->RequestDelete();
-}
-
 }  // namespace mozilla